components/region_cache_memory_engine/src/write_batch.rs

```diff
@@ -323,13 +324,45 @@ impl RangeCacheWriteBatchEntry {
         memory_controller: Arc<MemoryController>,
         guard: &epoch::Guard,
     ) -> Result<()> {
-        let handle = &skiplist_engine.data[self.cf];
-        let (mut key, mut value) = self.encode(seq);
-        key.set_memory_controller(memory_controller.clone());
-        value.set_memory_controller(memory_controller);
-        handle.insert(key, value, guard).release(guard);
+        let handle = skiplist_engine.cf_handle(id_to_cf(self.cf));
+        if self.cf == 1 && matches!(self.inner, WriteBatchEntryInternal::Deletion) {
```

Comment:
use an inline method is_lockcf(self.cf) instead. And 1 should be a const instead